lib/iris/config.py

```diff
@@ -154,3 +156,101 @@ def get_dir_option(section, option, default=None):
 
 
 IMPORT_LOGGER = get_option(_LOGGING_SECTION, 'import_logger')
+
+
+#################
+# Runtime options
+
+class NetCDF(object):
+    """Control Iris NetCDF options."""
+
+    def __init__(self, conventions_override=None):
+        """
+        Set up NetCDF processing options for Iris.
+
+        Currently accepted kwargs:
+
+        * conventions_override (bool):
+            Define whether the CF Conventions version (e.g. `CF-1.6`) set when
+            saving a cube to a NetCDF file should be defined by
+            Iris (the default) or the cube being saved.
+
+            If `False` (the default), specifies that Iris should set the
+            CF Conventions version when saving cubes as NetCDF files.
+            If `True`, specifies that the cubes being saved to NetCDF should
+            set the cf conventions version for the saved NetCDF files.
+
+        Example usages:
+
+        * Specify, for the lifetime of the session, that we want all cubes
+          written to NetCDF to define their own CF Conventions versions::
+
+            iris.options.netcdf.conventions_override = True
+            iris.save('my_cube', 'my_dataset.nc')
+            iris.save('my_second_cube', 'my_second_dataset.nc')
+
+        * Specify, with a context manager, that we want a cube written to
+          NetCDF to define its own CF Conventions version::
+
+            with iris.options.netcdf.context(conventions_override=True):
+                iris.save('my_cube', 'my_dataset.nc')
+
+        """
+        # Define allowed `__dict__` keys first.
+        self.__dict__['conventions_override'] = None
+
+        # Now set specific values.
+        setattr(self, 'conventions_override', conventions_override)
+
+    def __str__(self):
+        msg = 'NetCDF options: conventions_override={}.'
+        return msg.format(self.conventions_override)
+
+    def __setattr__(self, name, value):
+        if name not in self.__dict__:
+            # Can't add new names.
+            msg = "'Option' object has no attribute {!r}".format(name)
+            raise AttributeError(msg)
+        if value is None:
+            # Set an unset value to the name's default.
+            value = self._defaults_dict[name]['default']
+        if self._defaults_dict[name]['options'] is not None:
+            # Replace a bad value with a good one if there is a defined set of
+            # specified good values. If there isn't, we can assume that
+            # anything goes.
+            if value not in self._defaults_dict[name]['options']:
+                good_value = self._defaults_dict[name]['default']
+                wmsg = ('Attempting to set bad value {!r} for attribute {!r}. '
```

Comment:
Minor suggestion, `bad` -> `invalid`

Do we want to raise a warning here, which may be silently ignored and use the default, or raise a hard exception ...